feat: Adding custom default template #3658 (#4035)
gruntwork-io/terragrunt: docs/_docs/02_features/05-catalog.md

If `<repo-url>` is provided, the repository will be cloned into a temporary directory, otherwise:

> 1. The repository list are searched in the [config file](#configuration) `terragrunt.hcl`. if `terragrunt.hcl` does not exist in the current directory, the config are searched in the parent directories.

Reviewer: ```suggestion
1. The `urls` listed in the `catalog` configuration of your [parent configuration file](#configuration) file are used. If the root configuration file does not exist in the current directory, parent directories are recursively searched.
```

I know you didn't write most of this, but we're editing it here, so let's fix it.

Author: updated

1. If the repository list is not found in the configuration file, the modules are looked for in the current directory.

> The provided repositories will recursively be searched for OpenTofu/Terraform modules in the root of the repo and the `modules` directory. A table with all the discovered Terraform modules will subsequently be displayed. You can then:

Reviewer: ```suggestion
For each of the provided repositories, Terragrunt will recursively search for OpenTofu/Terraform modules from the root of the repo and the `modules` directory. A table with all the discovered OpenTofu/Terraform modules will subsequently be displayed.

You can then:
```

Author: updated

1. Search and filter the table: `/` and start typing.
1. Select a module in the table: use the arrow keys to go up and down and next/previous page.
1. See the docs for a selected module: `ENTER`.

> 1. Use [`terragrunt scaffold`]({{site.baseurl}}/docs/features/scaffold/) to render a `terragrunt.hcl` for using the module: `S`.

Reviewer: ```suggestion
1. Use [`terragrunt scaffold`]({{site.baseurl}}/docs/features/scaffold/) to render a `terragrunt.hcl` file that uses the module: `S`.
```

### Custom templates for scaffolding

> Terragrunt has a basic template built-in for rendering `terragrunt.hcl` files, but you can provide your own templates to customize what code is generated! Scaffolding is done via [boilerplate](https://github.com/gruntwork-io/boilerplate), and Terragrunt allows you to specify custom boilerplate templates via two mechanisms while using catalog:

Reviewer: ```suggestion
Terragrunt has a basic template built-in for rendering `terragrunt.hcl` files, but you can provide your own templates to customize how code is generated! Scaffolding is done via an integration with [boilerplate](https://github.com/gruntwork-io/boilerplate), and Terragrunt allows you to specify custom boilerplate templates via two mechanisms while using catalog:
```

Author: updated

> 1. You can define a custom boilerplate template in a `.boilerplate` subfolder of your Terraform module.

Reviewer: ```suggestion
1. You can define a custom Boilerplate template in a `.boilerplate` sub-directory of any OpenTofu/Terraform module.
```

Author: updated

> 1. You can specify a custom boilerplate template in the [catalog config](#configuration).

Reviewer: ```suggestion
1. You can specify a custom Boilerplate template in the [catalog config](#configuration).
```

Author: updated

### Configuration

> An example of how to define the optional default template, and the list of repositories for the `catalog` command in the `terragrunt.hcl` configuration file:

Reviewer: ```suggestion
An example of how to define the optional default template, and the list of repositories for the `catalog` command in a `root.hcl` configuration file:
```

Author: updated

``` hcl
catalog {

>   default_template = "git@github.com/gruntwork-io/terragrunt.git//test/fixture-scaffold/external-template"

Reviewer: ```suggestion
  default_template = "git@github.com/acme/example.git//path/to/template"
```

Author: updated

}
```

## Scaffolding Flags

The following `catalog` flags control behavior of the underlying `scaffold` command when the `S` key is pressed in a catalog entry:

- `--no-include-root` - Do not include the root configuration file in any generated `terragrunt.hcl` during scaffolding.
- `--root-file-name` - The name of the root configuration file to include in any generated `terragrunt.hcl` during scaffolding. This value also controls the name of the root configuration file to search for when trying to determine Catalog urls.

> - `--output-folder` - Location for generated `terragrunt.hcl`. If flag is not provided current working directory is selected.

Reviewer: ```suggestion
- `--output-folder` - Location for the scaffolded configurations. If flag is not provided current working directory is selected.
```

Author: updated
